codex-rs/tui/src/chatwidget.rs

```diff
@@ -1278,7 +1302,11 @@ impl ChatWidget {
 
     /// Set the model in the widget's config copy.
     pub(crate) fn set_model(&mut self, model: String) {
+        let header_changed = self.session_header.set_model(model.clone());
```

Comment:
I would change `model` to be `&str` instead of `String` and avoid `clone()` when it's not really necessary.